massa-execution-worker/src/tests/scenarios_mandatories.rs

```diff
@@ -986,188 +901,164 @@ fn send_and_receive_transaction() {
+    let finalized_waitpoint_trigger_handle = finalized_waitpoint.get_trigger_handle();
     let keypair = KeyPair::from_str(TEST_SK_1).unwrap();
     let address = Address::from_public_key(&keypair.get_public_key());
-    let mut foreign_controllers = ExecutionForeignControllers::new_with_mocks();
-    selector_boilerplate(
-        &mut foreign_controllers.selector_controller,
-        &block_producer,
-    );
-    let mut universe = ExecutionTestUniverse::new(foreign_controllers, exec_cfg.clone());
-
-    // get initial balance
-    let balance_initial = universe
-        .final_state
-        .read()
-        .ledger
-        .get_balance(&address)
-        .unwrap();
-
-    // get initial roll count
-    let roll_count_initial = universe
-        .final_state
-        .read()
-        .pos_state
-        .get_rolls_for(&address);
-    let roll_sell_1 = 10;
-    let roll_sell_2 = 1;
-    let roll_sell_3 = roll_count_initial.saturating_add(10);
-
-    let initial_deferred_credits = Amount::from_str("100").unwrap();
-
+    let (rolls_path, _) = get_initials();
     let mut batch = DBBatch::new();
+    let initial_deferred_credits = Amount::from_str("100").unwrap();
+    let mut pos_final_state = PoSFinalState::new(
+        PoSConfig::default(),
+        "",
+        &rolls_path.into_temp_path().to_path_buf(),
+        foreign_controllers.selector_controller.clone(),
+        foreign_controllers.db.clone(),
+    )
+    .unwrap();
+    pos_final_state.create_initial_cycle(&mut batch);
+    pos_final_state.put_deferred_credits_entry(
+        &Slot::new(1, 0),
+        &address,
+        &initial_deferred_credits,
+        &mut batch,
+    );
 
-    // set initial_deferred_credits that will be reimbursed at first block
-    universe
-        .final_state
-        .write()
-        .pos_state
-        .put_deferred_credits_entry(
-            &Slot::new(1, 0),
-            &address,
-            &initial_deferred_credits,
-            &mut batch,
-        );
-
-    universe
-        .final_state
-        .write()
+    foreign_controllers
         .db
         .write()
         .write_batch(batch, Default::default(), None);
+    pos_final_state.recompute_pos_state_caches();
+    final_state_boilerplate(
+        &mut foreign_controllers.final_state,
+        foreign_controllers.db.clone(),
+        &mut foreign_controllers.selector_controller,
+        &mut foreign_controllers.ledger_controller,
+        None,
+        None,
+        Some(pos_final_state),
+    );
+    foreign_controllers
+        .final_state
+        .write()
+        .expect_finalize()
+        .times(1)
+        .with(predicate::eq(Slot::new(3, 0)), predicate::always())
+        .returning(move |_, changes| {
+            let amount = changes
+                .ledger_changes
+                .get_balance_or_else(&address, || None)
+                .unwrap();
+            assert_eq!(
+                amount,
+                Amount::from_mantissa_scale(100, 0)
```

Comment:
Maybe specify that this is coming from the boilerplate, or set the default boilerplate value to 0